pkg/instance/repository.go

```diff
@@ -21,6 +21,34 @@ type repository struct {
 	instanceParameterEncryptionKey string
 }
 
+func (r repository) SaveDeployment(deployment *model.Deployment) error {
+	// TODO: Do we need the option to save nested entities... Yes, if we create the deployment from a preset we need to store all the links as well
+	//err := r.db.Session(&gorm.Session{FullSaveAssociations: true}).Save(deployment).Error
+	err := r.db.Create(&deployment).Error
+	if errors.Is(err, gorm.ErrDuplicatedKey) {
+		return errdef.NewDuplicated("a deployment named %q already exists", deployment.Name)
+	}
+
+	return err
```

Comment:
Some time back we discussed that we would provide our own errors instead of just returning 3rd party errors. https://github.com/dhis2-sre/im-doc/blob/master/dev/README.md#errors

I would at least do `fmt.Errorf("failed to save deployment: %v", err)`. In the long run we should log details with a correlation id and only return a high level error (explanation) to the user. This applies to all repository methods.